generic_relations/relations.py

```diff
@@ -83,8 +90,12 @@ def determine_serializer_for_data(self, value):
                 serializer.from_native(value)
                 # Collects all serializers that can handle the input data.
                 serializers.append(serializer)
-            except:
-                pass
+            except ValidationError as e:
+                if self.determining_errors and e.message not in self.determining_errors:
+                    # allow validation error to bubble up
+                    serializers.append(serializer)
+                else:
+                    pass
```

Comment:
I'm thinking that this could look something like:

```python
serializers = []
errors = []
for model, serializer in six.iteritems(self.serializers):
    try:
        serializer.from_native(value)
    except ValidationError as error:
        model_name = model.Meta.verbose_name
        errors.append({model_name: error.message})
    else:
        serializers.append(serializer)
```

I'm not certain `model.Meta.verbose_name` is correct, but the rest should work.